pkg/authentication/basic/htpasswd.go

```diff
@@ -5,18 +5,21 @@ import (
 	"crypto/sha1" // #nosec G505
 	"encoding/base64"
 	"encoding/csv"
-	"fmt"
 	"io"
 	"os"
+	"sync/atomic"
+	"unsafe"
 
 	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/logger"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/watcher"
 	"golang.org/x/crypto/bcrypt"
 )
 
-// htpasswdMap represents the structure of an htpasswd file.
+// HtpasswdMap represents the structure of an htpasswd file.
 // Passwords must be generated with -B for bcrypt or -s for SHA1.
-type htpasswdMap struct {
-	users map[string]interface{}
+type HtpasswdMap struct {
+	usersFile string
+	m         unsafe.Pointer
```

Comment:
Have you considered using a map still and using a mutex when updating it to prevent concurrency issues?

Reply:
Changed back to using a map and removed the unsafe pointer.